src/arbitrum/utils/YieldOptimizersRegistry.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// ███████╗░█████╗░██████╗░████████╗██████╗░███████╗░██████╗░██████╗
+// ██╔════╝██╔══██╗██╔══██╗╚══██╔══╝██╔══██╗██╔════╝██╔════╝██╔════╝
+// █████╗░░██║░░██║██████╔╝░░░██║░░░██████╔╝█████╗░░╚█████╗░╚█████╗░
+// ██╔══╝░░██║░░██║██╔══██╗░░░██║░░░██╔══██╗██╔══╝░░░╚═══██╗░╚═══██╗
+// ██║░░░░░╚█████╔╝██║░░██║░░░██║░░░██║░░██║███████╗██████╔╝██████╔╝
+// ╚═╝░░░░░░╚════╝░╚═╝░░╚═╝░░░╚═╝░░░╚═╝░░╚═╝╚══════╝╚═════╝░╚═════╝░
+// ███████╗██╗███╗░░██╗░█████╗░███╗░░██╗░█████╗░███████╗
+// ██╔════╝██║████╗░██║██╔══██╗████╗░██║██╔══██╗██╔════╝
+// █████╗░░██║██╔██╗██║███████║██╔██╗██║██║░░╚═╝█████╗░░
+// ██╔══╝░░██║██║╚████║██╔══██║██║╚████║██║░░██╗██╔══╝░░
+// ██║░░░░░██║██║░╚███║██║░░██║██║░╚███║╚█████╔╝███████╗
+// ╚═╝░░░░░╚═╝╚═╝░░╚══╝╚═╝░░╚═╝╚═╝░░╚══╝░╚════╝░╚══════╝
+
+//  _____         _                   _____     _   _ _____         _     _           
+// |   __|___ ___| |_ ___ ___ ___ ___|  _  |___| |_|_| __  |___ ___|_|___| |_ ___ _ _ 
+// |   __| . |  _|  _|  _| -_|_ -|_ -|     |  _| . | |    -| -_| . | |_ -|  _|  _| | |
+// |__|  |___|_| |_| |_| |___|___|___|__|__|_| |___|_|__|__|___|_  |_|___|_| |_| |_  |
+//                                                             |___|             |___|
+
+// Github - https://github.com/FortressFinance
+
+contract YieldOptimizersRegistry {
+
+    enum AMMType {
+        Curve,
+        Balancer
+    }
```

Comment:
Does that structure mean we are limiting our functionality strictly to these protocols? Would this then mean we had to redeploy contracts in order to extend our product suit? Following from that, would there be a more dynamic approach you can think of?

Reply:
Yes. I'm thinking that's the correct approach from a strategic POV, we will be able to offer a bunch of strategies and those 2 are the best AMMs out there IMO, then we should focus on our other products.

Also it's only a registry, we can always redeploy...